### PR TITLE
Allow to share proxy via QR

### PR DESCRIPTION
**Pull request by adbenitez**
add new API DcContext.createQrSvg() and use it to generate proxy QRs for sharing

image

### CODE REVIEW

File: jni/dc_wrapper.c [hunk 1]
```diff
@@ -884,6 +884,16 @@ JNIEXPORT jstring Java_com_b44t_messenger_DcContext_getSecurejoinQrSvg(JNIEnv *e
     return ret;
 }
 
+JNIEXPORT jstring Java_com_b44t_messenger_DcContext_createQrSvg(JNIEnv *env, jobject obj, jstring payload)
+{
+    CHAR_REF(payload);
+        char* temp = dc_create_qr_svg(payloadPtr);
+    CHAR_UNREF(payload);
+    jstring ret = JSTRING_NEW(temp);
+    dc_str_unref(temp);
```
Comment (r10s) [inline jni/dc_wrapper.c:893]: unreferencing in the opposite order as referencing matches better the style used elsewhere:

```suggestion
        jstring ret = JSTRING_NEW(temp);
        dc_str_unref(temp);
    CHAR_UNREF(payload);
```

**Review by r10s — APPROVED** [review]
lgtm, did not test, but if it crashes, we will notice immediately :)

**Comment by adbenitez** [comment]
> but if it crashes

this PR depends in your core PR that got recently merged, new core is needed before this can be merged

File: src/main/res/layout/dialog_share_proxy.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ScrollView
```
Comment (Hocuri) [inline src/main/res/layout/dialog_share_proxy.xml:2]: Why a ScrollView here?

Comment (adbenitez) [reply src/main/res/layout/dialog_share_proxy.xml:2]: general precaution but yeah it is unlikely that this screen could ever get so big, could remove it

Comment (adbenitez) [reply src/main/res/layout/dialog_share_proxy.xml]: I removed it :+1:

**Review by r10s — APPROVED** [review]

File: src/main/res/values/strings.xml [hunk 3]
```diff
@@ -621,12 +621,14 @@
     <string name="proxy_use_proxy">Use Proxy</string>
     <string name="proxy_add">Add Proxy</string>
     <string name="proxy_add_explain">Supported proxy types: HTTP(S), SOCKS5 and Shadowsocks.</string>
-    <string name="proxy_add_url_hint">Enter proxy URL here</string>
-    <string name="proxy_invalid">Invalid or unsupported proxy URL</string>
+    <string name="proxy_add_url_hint">Enter proxy link here</string>
+    <string name="proxy_invalid">Invalid or unsupported proxy link</string>
```
[the remark this answers is not in the document]
Comment (adbenitez) [reply src/main/res/values/strings.xml:625]: I think link is a bit more user friendly than URL, and we already use "link" for other URL like the invite links, so it is easier to know what a link is then in this screen, Telegram also uses link:
image

**Review by Hocuri — APPROVED** [review]
Didn't test, but LGTM

**Comment by r10s** [comment]
updated the core on main; this PR needs a rebase

**Pull request merged**